source/initial_composition/slab_model.cc

```diff
@@ -0,0 +1,123 @@
+/*
+  Copyright (C) 2011 - 2023 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include <aspect/initial_composition/slab_model.h>
+#include <aspect/utilities.h>
+
+
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    template <int dim>
+    void
+    SlabModel<dim>::initialize ()
+    {
+      // The input slabs are defined from the surface of the model
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(this->get_geometry_model().translate_symbolic_boundary_name_to_id("top"));
+
+      // The two columns correspond to slabs depth and thickness
+      slab_boundary.initialize(surface_boundary_set, 2);
```

Comment:
Please make the same change here that I did in #5028 for the other places where this operation happens.